apiserver/discoverspaces/discoverspaces.go

```diff
@@ -1,154 +0,0 @@
-// Copyright 2015 Canonical Ltd.
-// Licensed under the AGPLv3, see LICENCE file for details.
-
-package discoverspaces
-
-import (
-	"github.com/juju/errors"
-	"github.com/juju/utils/set"
-	"gopkg.in/juju/names.v2"
-
-	"github.com/juju/juju/apiserver/common"
-	"github.com/juju/juju/apiserver/common/networkingcommon"
-	"github.com/juju/juju/apiserver/facade"
-	"github.com/juju/juju/apiserver/params"
-	"github.com/juju/juju/network"
-	"github.com/juju/juju/state"
-)
-
-// API implements the API used by the discoverspaces worker.
-type API struct {
-	st         networkingcommon.NetworkBacking
-	resources  facade.Resources
-	authorizer facade.Authorizer
-}
-
-// NewAPI creates a new instance of the DiscoverSpaces API.
-func NewAPI(st *state.State, resources facade.Resources, authorizer facade.Authorizer) (*API, error) {
-	return NewAPIWithBacking(networkingcommon.NewStateShim(st), resources, authorizer)
-}
-
-// NewAPIWithBacking creates an API instance from the given network
-// backing (primarily useful from tests).
-func NewAPIWithBacking(st networkingcommon.NetworkBacking, resources facade.Resources, authorizer facade.Authorizer) (*API, error) {
-	if !authorizer.AuthController() {
```

Comment:
So this seems to be that no agents can ever call this, which should mean it is ok to remove. (we don't ever support running heterogenous controllers even if we do support agents not being in lock step)